lib/src/arithmetic/sign_magnitude_adder.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// sign_magnitude_adder.dart
+// Implementation of a One's Complement Adder
+//
+// 2024 August 8
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// An Adder which performs one's complement arithmetic using an
+/// adder that is passed in using a functor. If the caller can guarantee
+/// that the larger magnitude or negative value is provided first in 'a', then
+/// they can set 'largestMagnitudeFirst' to 'true' to avoid a comparator.
+class SignMagnitudeAdder extends Adder {
+  /// The sign of the first input
+  @protected
+  late final Logic aSign;
+
+  /// The sign of the second input
+  @protected
+  late final Logic bSign;
+
+  /// The sign of the result
+  Logic get sign => output('sign');
+
+  /// Largest magnitude argument is provided in [a] or if equal
+  /// the argument with a negative sign.
+  bool largestMagnitudeFirst;
+
+  /// [SignMagnitudeAdder] constructor with an unsigned adder functor
+  SignMagnitudeAdder(Logic as, super.a, Logic bs, super.b,
+      Adder Function(Logic, Logic) adderGen,
+      {this.largestMagnitudeFirst = false})
+      : super(
+            name: 'Ones Complement Adder: '
+                '${adderGen.call(Logic(), Logic()).name}') {
+    aSign = addInput('aSign', as);
+    bSign = addInput('bSign', bs);
+    final sign = addOutput('sign');
+
+    final bLarger = a.lt(b) | (a.eq(b) & bSign.gt(aSign));
+
+    final computeSign = mux(largestMagnitudeFirst ? Const(1) : Const(0), aSign,
```

Comment:
You can use the dart ternary operator directly for generation-time stuff, and `mux` for hardware.

I think this could just be 
```dart
    final computeSign = largestMagnitudeFirst ? aSign : mux(bLarger, bSign, aSign);
```

You could also maybe avoid creation of the `bLarger` logic by using more of an `if`/`else` block here instead, even though it would synthesize ok you might end up with extra SV and extra simulation cost (small).

Reply:
I fixed this.  There are cases where it is a little harder:  see my new onesComplementAdder which takes either a Logic subtractIn or a boolean subtract.  It's hard to avoid converting the boolean to a signal to help.  But this is an interesting case:  if we want either a runtime configuration option or a compile time option that do the same thing.